os/drivers/bluetooth/virtual_ble/vble_handler.c

```diff
@@ -0,0 +1,472 @@
+    case LWNL_EVT_BLE_CLIENT_CONNECT:
+        vble_connected_list.connected_count++;
+        client_init_parm.trble_device_connected_cb(&vble_connect);
+        break;
+    case LWNL_EVT_BLE_CLIENT_DISCONNECT:
+        vble_connected_list.connected_count--;
+        vble_connected_list.conn_handle[vble_connect.conn_handle] = VBLE_NO_CONNECTION;
+        client_init_parm.trble_device_disconnected_cb(vble_connect.conn_handle);
+        break;
+    case LWNL_EVT_BLE_CLIENT_NOTI:
+        client_init_parm.trble_operation_notification_cb(&vble_op_handle, &vble_op_data);
+        break;
+    case LWNL_EVT_BLE_SCAN_STATE:
+        client_init_parm.trble_scan_state_changed_cb(result);
+        if (result == TRBLE_SCAN_STOPPED) {
+            break;
+        }
+        if (g_filter.whitelist_enable != 0) {
+            int i;
+            for (i = 0; i < SCAN_WHITELIST_SIZE; i++) {
+                if (g_scan_whitelist[i].use != SCAN_WHITELIST_EMPTY) {
+                    memcpy(vble_scanned_items.addr.mac, g_scan_whitelist[i].addr.mac, TRBLE_BD_ADDR_MAX_LEN);
+                    client_init_parm.trble_device_scanned_cb(&vble_scanned_items);
+                }
+            }
+        } else {
+            uint8_t dummy[] = {0x00, 0x1E, 0xC0, 0x04, 0x9F, 0xF3};
+            memcpy(vble_scanned_items.addr.mac, dummy, TRBLE_BD_ADDR_MAX_LEN);
+            client_init_parm.trble_device_scanned_cb(&vble_scanned_items);
+            vble_scanned_items.addr.mac[5] = 0xF4;
+            client_init_parm.trble_device_scanned_cb(&vble_scanned_items);
+            vble_scanned_items.addr.mac[5] = 0xF5;
+            client_init_parm.trble_device_scanned_cb(&vble_scanned_items);
+        }
+        break;
+    }
+}
+
+static void _generate_evt(int argc, char *argv[])
+{
+    lwnl_cb_status event_type = {LWNL_DEV_BLE, 0};
+    event_type.evt = atoi(argv[1]);
+    int result = atoi(argv[2]);
+    int sleep_time = atoi(argv[3]);
+
+    VBLE_LOG("[VBLE] sleep (%d) result (%d)event type(%d)\n",
+                 sleep_time, result, event_type.evt);
+
+    sleep(sleep_time);
+    vble_callback_handler(event_type, result);
+}
+
+int vble_create_event(uint32_t event, int32_t result, int32_t sleep)
+{
+    char event_buf[16] = {0,};
+    char res_buf[16] = {0,};
+    char sleep_buf[16] = {0,};
+
+    memset(event_buf, 0, 16);
+    memset(res_buf, 0, 16);
+    memset(sleep_buf, 0, 16);
+
+    snprintf(event_buf, 16, "%d", event);
+    snprintf(res_buf, 16, "%d", result);
+    snprintf(sleep_buf, 16, "%d", sleep);
+
+    char *argv[4] = {0,};
+    argv[0] = event_buf;
+    argv[1] = res_buf;
+    argv[2] = sleep_buf;
+    argv[3] = NULL;
+    int res = kernel_thread("vble_evt", 100, 2048, _generate_evt, argv);
+    if (res == -1) {
+        VBLE_ERROR(0);
+        return -1;
+    }
+    return 0;
+}
+
+static void init_handler(struct vconfigs *config) {
+    //init client
+    client_init_parm.trble_scan_state_changed_cb = config->client->trble_scan_state_changed_cb;
```

Comment:
Is it okay not to check that `config` is NULL or not?

Reply:
It is checked beforehand by BLE manager